moped-editor/src/views/staff/StaffForm.js

```diff
@@ -59,33 +74,51 @@ const statuses = [
   { value: "0", name: "Inactive" },
 ];
 
-const staffValidationSchema = yup.object().shape({
-  first_name: yup.string().required(),
-  last_name: yup.string().required(),
-  title: yup.string().required(),
-  workgroup: yup.string().required(),
-  workgroup_id: yup.string().required(),
-  email: yup.string().required(),
-  password: yup.string().required(),
-  roles: yup.string().required(),
-  status_id: yup.string().required(),
-});
+// Pass editFormData to conditionally validate if adding or editing
+const staffValidationSchema = editFormData =>
+  yup.object().shape({
+    first_name: yup.string().required(),
+    last_name: yup.string().required(),
+    title: yup.string().required(),
+    workgroup: yup.string().required(),
+    workgroup_id: yup.string().required(),
+    email: yup.string().required(),
+    password: yup.mixed().when({
+      // If we are editing a user, password is optional
+      is: () => editFormData === null,
+      then: yup.string().required(),
+      otherwise: yup.string(),
+    }),
```

Comment:
Ha! This is brilliant! 🙌

Reply:
Thanks! It was nice to find a way to make the validations conditional.